Rewrite of the docs (draft) (#40)
fox-it/dissect-docs: docs/source/index.rst

Quick Demo
----------

In just a couple of seconds you can query a forensic image or even explore it
using **target-shell**!

.. image:: /images/demo.gif

Usage Example
-------------

One of the most prominent tools that Dissect offers is called **target-query**.
With this simple one-liner you can extract all user accounts from an image: 

.. code-block:: console

    $ target-query pc.img -f users
    
    <windows/user ... name='systemprofile' ...>
    <windows/user ... name='LocalService' ...>
    <windows/user ... name='NetworkService' ...>
    <windows/user ... name='Mr. Evil' ...>
    
To see what other useful artefacts you can query use ``-l``:

.. code-block:: console

    $ target-query pc.img -l

This will list all Dissect functions that are available for this image.
For a complete list of all functions, regardless of the image, just omit the image argument.

Key features
------------

Dissect is a powerful artefact extractor and parser that saves you a lot of time.
With Dissect you can:

* Quickly **extract** artefacts from any source (like IMG, EWF, Kape, DD, VDI, PVM)
* Access artefacts from almost any **OS** (like Windows, macOS, Linux/Unix, ESXi)

> * Access almost any **filesystem** (like NTFS, EXT, XFS, QNX6)

Reviewer: I believe the feature is still in review. At least adding the support into `dissect.target`.

* **Parse** (binary) logs, registry entries, cookies, history and more, directly from the source
* **Export** findings to text, JSON, CSV or stream to datastores like Splunk, Elastic, MySQL
* **Feed, filter and script** to process found data in any way you like

> * Investigate images with powerful tools like **target-shell**, **target-mount** or the **Python API**

Reviewer: Any reason why the full stops are left out in this list?

> Using these features, you can for instance (among other things):
> 
> * Make a timeline of the event log from all investigation data at once
> * Identify anomalies in artefacts such as services, tasks and run keys over a large amount of investigation data
> * Perform incident response on partially ransomwared (encrypted) virtual machine disks
> * Perform complex IOC checks on thousands of hosts in a couple of hours
> * Export all USN journal records from a Bitlocker-encrypted disk to Splunk without waiting for decryption
> * Collect forensic artefacts from all live virtual machines directly from the hypervisor with no down time
> * Export all artefacts easily to any data format you want, for example CSV, JSON or Avro, or stream directly to Splunk or Elastic

Reviewer: ```suggestion
Using these features, you can for instance (among other things):

* Make a **timeline** of the event log from all your investigation data at once.
* Identify **anomalies** in artefacts such as services, tasks and run keys over a large amount of investigation data.
* Do incident response on partially **ransomwared** (encrypted) virtual machine disks.
* Extract artefacts from **LVM, RAID** and other disk setups, without any manual configuration.
* Perform complex **IOC checks** on thousands of hosts in a couple of hours.
* Collect forensic artefacts from all **live** virtual machines directly from the hypervisor with no down time.
* Export all artefacts easily to any data format you want, for example CSV, JSON or Avro, or stream directly to Splunk or Elastic.
```
Removed this line since this feature is not yet in the public version.
>* Export all USN journal records from a Bitlocker-encrypted disk to Splunk without waiting for decryption.

> Easy to setup

Reviewer: ```suggestion
Easy to install
```

> Easy to install
> -------------

Reviewer: ```suggestion
Easy to install
---------------
```

Dissect can be installed using pip:

.. code-block:: console

    $ pip install dissect

It is recommended that you use a `virtual environment <https://docs.python.org/3/tutorial/venv.html>`_.

.. note ::
    .. include:: /versions.rst
    
You can also try dissect directly from within your browser:

`try.dissect.tools <https://try.dissect.tools>`_
    

Setup with Docker
-----------------

> If you simply want to get started with some of the examples without having to install anything, a basic Docker image

Reviewer: If we have examples its good to link them here!

is available `here <https://github.com/orgs/fox-it/packages/container/package/dissect>`_ (`mirror <https://hub.docker.com/r/dissect/dissect>`_).
You can start using this image by executing the following command in your terminal:
 
.. code-block:: console

>     $ docker run -it --rm -v /path/to/targets/:/mnt:ro ghcr.io/fox-it/dissect:3.11
>     (<dissect version>) <container hash>:/workspace$

Reviewer: ```suggestion
    $ docker run -it --rm -v /path/to/targets/:/mnt:ro ghcr.io/fox-it/dissect:3.11
    (<dissect version>) <container hash>:/workspace$ target-query -l
```

Learn more
----------

Excited? Great, so are we!
The next step is to get to know Dissect a little better.

* Visit :doc:`/usage/introduction` for an in-depth introduction into Dissect
* Visit :doc:`/tools/index` for an overview of each tool in the Dissect suite

Or you can start by taking a look at some community articles and videos:
:doc:`/resources/dissect-in-action` or

> :doc:`/resources/talks-and-conferences` to begin with.

Reviewer: Adding a little bit of how to get in touch with us on this page might be a good idea. (or linking to it somewhere here).

Welcome to Dissect!
===================

Effortlessly investigate **forensic artefacts** from any source with **Dissect**.
With Dissect, you can go from intake call to patient zero in a matter of hours,
even in infrastructures with thousands of systems.

Quick Demo
----------

In just a couple of seconds you can query a forensic image or even **shell-in**!
This short animation shows how to list deleted files from an suspicious machine

> and explore the contents of the machine through a shell...

Author: The animation got mixed up when I tried to convert it to asciinema, it's meant to show the recyclebin.

.. image:: /images/demo.gif

Usage Example
-------------

One of the most prominent tools that Dissect offers is called **target-query**.

> With this simple one-liner you can extract all user accounts from an image: 

Author: Should be plural I guess?

.. code-block:: console

    $ target-query pc.img -f users
    
    <windows/user ... name='systemprofile' ...>
    <windows/user ... name='LocalService' ...>
    <windows/user ... name='NetworkService' ...>
    <windows/user ... name='Mr. Evil' ...>

Key features
------------

Dissect is a powerful artefact extractor and parser that saves you a lot of time.
With Dissect you can:

* Quickly **extract** artefacts from any source (like EWF, Kape, DD, VDI, PVM)
* Access artefacts from almost any **OS** (like Windows, macOS, Linux/Unix, ESXi)

> * Access almost any **filesystem** (like NTFS, EXT, XFS, QNX6)

Author: Isn't that covered by the the source formats?

* **Parse** (binary) logs, registry entries, cookies, history and more
* **Export** findings to Text, JSON, CSV or stream to datastores like Splunk, Elastic, MySQL
* **Feed, filter and script** to process found records in any way you like
* Investigate images with powerful tools like **target-shell**, **target-mount** or the **Python API**

Easy to setup
-------------

No matter your skill level, we have an easy setup solution for you!

> * Just install with `pip install dissect` from `PyPI <https://pypi.python.org/pypi/dissect>`_

Author: Installation is under the installation topic. That seems to make sense? I think it's better to focus on the core, i.e. how to use it on the frontpage. The fact that in can be installed using pip is not really a special thing right? I agree that it is difficult to decide what is important enough for the frontpage and what not, but hard choices must be made I guess.

* Try in the browser `try.dissect.tools <https://try.dissect.tools>`_
* Use the Docker image `here <https://github.com/orgs/fox-it/packages/container/package/dissect>`_ (`mirror <https://hub.docker.com/r/dissect/dissect>`_).

> About this manual

Author: According to the Cambridge dictionary (https://dictionary.cambridge.org/dictionary/english/manual), a manual is:

`a book that gives you practical instructions on how to do something or how to use something, such as a machine`

Given that we are in a digital era, I think the term fits. I mean, the documentation of various GNU/Linux tools is also provided as "man"(ual) pages? I believe some project including OpenBSD even serve man pages as part of their website (https://man.openbsd.org/man8/radiusd.8). BTW, they also call it Manual I believe? I would not know of a better term. You could also use the word "documentation", but that sounds like it was an "afterthought", i.e. some notes put together. I mean, that is not what this is? We dedicate a lot of time and effort to these pages, as such I suppose it is worthy of the term "manual".

> About this manual

Reviewer 2: A manual implies to me that you will describe step by step how to use Dissect to solve an IR case, which is not the case. These pages contain a lot more than instructions, so I'm not sure if calling it a manual is fair against the rest of the documentation.
